pyannote/core/annotation.py

```diff
@@ -472,6 +473,85 @@ def crop(self, support: Support, mode: CropMode = 'intersection'):
             else:
                 raise NotImplementedError("unsupported mode: '%s'" % mode)
 
+    def truncate(self, removed: Support, mode: CropMode = 'intersection') \
+            -> 'Annotation':
+        """Removes all segments or parts of segments that
+        overlap the `removed` argument.
+
+        Parameters
+        ----------
+        support : Segment or Timeline
+            If `support` is a `Timeline`, its support is used.
+        mode : {'strict', 'loose', 'intersection'}, optional
+            Controls how segments that are not fully included in `removed` are
+            handled. 'strict' mode only removes fully included segments. 'loose'
+            mode removes any intersecting segment. 'intersection' mode removes any
+            intersecting segment but replace them by their actual intersection.
+
+        Returns
+        -------
+        truncated : Annotation
+            Truncated annotation
+
+        Note
+        ----
+        In 'intersection' mode, the best is done to keep the track names
+        unchanged. However, in some cases where two original segments are
+        cropped into the same resulting segments, conflicting track names are
+        modified to make sure no track is lost.
+
+        """
+        if isinstance(removed, Segment):
+            removed = Timeline([removed])
+
+        extent_tl = Timeline([self.get_timeline().extent()], uri=self.uri)
+        truncating_support = removed.gaps(support=extent_tl)
+        # loose for truncate means strict for crop and vice-versa
+        if mode == "loose":
+            mode = "strict"
+        elif mode == "strict":
+            mode = "loose"
+        return self.crop(truncating_support, mode=mode)
+
+    def overlaps(self, for_labels: Optional[List[Label]] = None) -> 'Timeline':
```

Comment:
I read `overlaps` as the the third person of verb `to overlap`.
But, after reading the docstring, I understand that you mean the plural of noun `overlap`. 

We should find a better name. What about `to_overlap` or `get_overlap`? Or `overlap` without the `s`?

Also, I'd rather go with `labels` instead of `for_labels`, like what is done in `Annotation.subset`.

Reply:
I settled for `get_overlap`. Done for `for_labels`